pkg/controllers/lb/controller.go

```diff
@@ -467,3 +470,57 @@ func IsPorterService(obj runtime.Object) bool {
 	}
 	return false
 }
+
+// +kubebuilder:webhook:path=/validate-network-kubesphere-io-v1alpha2-svc,mutating=true,sideEffects=NoneOnDryRun,failurePolicy=fail,groups="",resources=services,verbs=create,versions=v1,name=mutating.eip.network.kubesphere.io
+
+type SvcAnnotator struct {
+	client.Client
+	decoder *admission.Decoder
+}
+
+func (r *SvcAnnotator) InjectDecoder(d *admission.Decoder) error {
+	r.decoder = d
+	return nil
+}
+
+func (r *SvcAnnotator) Handle(ctx context.Context, req admission.Request) admission.Response {
+	svc := &corev1.Service{}
+
+	if err := r.decoder.Decode(req, svc); err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+	if svc.Spec.Type != corev1.ServiceTypeLoadBalancer {
+		marshaledSvc, err := json.Marshal(svc)
+		if err != nil {
+			return admission.Errored(http.StatusInternalServerError, err)
+		}
+		return admission.PatchResponseFromRaw(req.Object.Raw, marshaledSvc)
+	}
+	// check default eip
+	eips := networkv1alpha2.EipList{}
+	err := r.List(context.Background(), &eips)
+	if err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+	for _, eip := range eips.Items {
+		if validate.HasPorterDefaultEipAnnotation(eip.Annotations) {
+			// exist default eip,injection annotation
+			if svc.Annotations == nil {
+				svc.Annotations = make(map[string]string)
+				svc.Annotations[constant.PorterAnnotationKey] = constant.PorterAnnotationValue
+			} else if value, ok := svc.Annotations[constant.PorterAnnotationKey]; !ok || value != constant.PorterAnnotationValue {
```

Comment:
If this values exists and not equal to `porter`, maybe we should not process this service.